src/Datadog.Trace.ClrProfiler.Managed/Emit/MethodBuilder.cs

```diff
@@ -313,36 +313,37 @@ private TDelegate EmitDelegate()
                                          .ToArray();
             }
 
-            var dynamicMethod = Emit<TDelegate>.NewDynamicMethod(methodInfo.Name);
+            DynamicMethod dynamicMethod = new DynamicMethod(methodInfo.Name, returnType, delegateParameterTypes, ObjectExtensions.Module, skipVisibility: true);
+            ILGenerator il = dynamicMethod.GetILGenerator();
 
             // load each argument and cast or unbox as necessary
             for (ushort argumentIndex = 0; argumentIndex < delegateParameterTypes.Length; argumentIndex++)
             {
                 Type delegateParameterType = delegateParameterTypes[argumentIndex];
                 Type underlyingParameterType = effectiveParameterTypes[argumentIndex];
 
-                dynamicMethod.LoadArgument(argumentIndex);
+                il.Emit(OpCodes.Ldarg, argumentIndex);
```

Comment:
this implementation misses the shortcuts from Ldarg_0 to Ldarg_3 and the short version Ldarg_S. I don't know if there is a performance impact by using the int16 version directly.